ui/components/multichain/pages/gator-permissions/components/review-gator-permission-item.tsx

```diff
@@ -42,17 +42,72 @@ import {
 } from '../../../../../../shared/lib/gator-permissions';
 import { PreferredAvatar } from '../../../../app/preferred-avatar';
 import { BackgroundColor } from '../../../../../helpers/constants/design-system';
-import {
-  getNativeTokenInfo,
-  selectERC20TokensByChain,
-} from '../../../../../selectors/selectors';
-import { getTokenMetadata } from '../../../../../helpers/utils/token-util';
 import { getPendingRevocations } from '../../../../../selectors/gator-permissions/gator-permissions';
+import { useGatorPermissionTokenInfo } from '../../../../../hooks/gator-permissions/useGatorPermissionTokenInfo';
+import { CopyIcon } from '../../../../app/confirm/info/row/copy-icon';
+import { Skeleton } from '../../../../component-library/skeleton';
+
+// Shared row style for permission details
+const rowStyle = { flex: '1', alignSelf: 'center' } as const;
 
-type TokenMetadata = {
-  symbol: string;
-  decimals: number | null;
-  name: string;
+type PermissionDetailRowProps = {
+  label: string;
+  value: React.ReactNode;
+  testId?: string;
+  isLoading?: boolean;
+};
+
+/**
+ * Reusable row component for permission details
+ *
+ * @param options - The component options
+ * @param options.label - The label text to display
+ * @param options.value - The value to display
+ * @param options.testId - Optional test ID for the value element
+ * @param options.isLoading - Whether to show a loading skeleton
+ * @returns The permission detail row component
+ */
+const PermissionDetailRow = ({
+  label,
+  value,
+  testId,
+  isLoading = false,
+}: PermissionDetailRowProps): JSX.Element => {
+  return (
+    <Box
+      flexDirection={BoxFlexDirection.Row}
+      justifyContent={BoxJustifyContent.Between}
+      style={rowStyle}
+      gap={4}
+      marginTop={2}
+    >
+      <Text
+        textAlign={TextAlign.Left}
+        color={TextColor.TextAlternative}
+        variant={TextVariant.BodyMd}
+      >
+        {label}
+      </Text>
+      <Box
+        flexDirection={BoxFlexDirection.Row}
+        justifyContent={BoxJustifyContent.End}
+        style={rowStyle}
+        gap={2}
+        alignItems={BoxAlignItems.Center}
+      >
+        <Skeleton isLoading={isLoading} width="100px" height="16px">
```

Comment:
Oh this is an interesting paradigm - I always just switched out between a skeleton view and a content view. Nice!